### PR TITLE
getTemporaryFilesByUserId function added to TemporaryFileDao

### PR DESCRIPTION
**Pull request by MrRob100**

### CODE REVIEW

File: classes/file/TemporaryFileDAO.inc.php [hunk 1]
```diff
@@ -34,6 +34,18 @@ function getTemporaryFile($fileId, $userId) {
 		$row = (array) $result->current();
 		return $row?$this->_returnTemporaryFileFromRow($row):null;
 	}
+	
+	/**
+	 * Retrieve temporary files by user Id
+	 * @param $userId int
+	 * @return Generator
+	 */
+	public function getTemporaryFilesByUserId($userId) {
```
Comment (asmecher) [inline classes/file/TemporaryFileDAO.inc.php:43]: Could you modify this function to return a `DAOResultFactory` invoking `_returnTemporaryFileFromRow`? It should be...
```php
$result = $this->retrieve(
	'SELECT * FROM temporary_files WHERE user_id = ? ORDER BY date_uploaded DESC LIMIT 1',
	[(int) $userId]
);
return new DAOResultFactory($result, $this, '_returnTemporaryFileFromRow');
```
That way this function takes care of instantiating the `TemporaryFile` objects, and calling code doesn't need to know about the raw database columns.

File: classes/file/TemporaryFileDAO.inc.php [hunk 2]
```diff
@@ -34,6 +34,18 @@ function getTemporaryFile($fileId, $userId) {
...
+		return $this->retrieve(
+			'SELECT * FROM temporary_files WHERE user_id = ? ORDER BY date_uploaded DESC LIMIT 1',
```
Comment (asmecher) [inline classes/file/TemporaryFileDAO.inc.php:45]: The `LIMIT 1` probably shouldn't be here!

File: classes/file/TemporaryFileDAO.inc.php (hunk 1 above)
Comment (MrRob100) [reply classes/file/TemporaryFileDAO.inc.php:43]: @asmecher Okay was there a way of doing it not using ORDER BY in the DAO method?

Comment (asmecher) [reply classes/file/TemporaryFileDAO.inc.php:43]: I'm afraid I'm not following -- can you write a little more?

Comment (MrRob100) [reply classes/file/TemporaryFileDAO.inc.php:43]: Your suggestion 
`$result = $this->retrieve(
	'SELECT * FROM temporary_files WHERE user_id = ?',
	[(int) $userId]
);
return new DAOResultFactory($result, $this, '_returnTemporaryFileFromRow');`

didn't have the `ORDER BY` in the statement which I have added, I wondered if you had removed it for a reason

`		$result = $this->retrieve(
			'SELECT * FROM temporary_files WHERE user_id = ? ORDER BY date_uploaded DESC',
			[(int) $userId]
		);
		return new DAOResultFactory($result, $this, '_returnTemporaryFileFromRow');`

Comment (asmecher) [reply classes/file/TemporaryFileDAO.inc.php:43]: No objection to having the `ORDER BY` coded into the query!

**Pull request merged**